src/Psalm/Internal/Type/TypeParser.php

```diff
@@ -1468,8 +1468,12 @@ private static function getTypeFromKeyedArrayTree(
                     if ($const_name === 'class') {
                         $property_key = $fq_classlike_name;
                         $class_string = true;
-                    } else {
+                    } elseif ($property_branch->value[0] === '"' || $property_branch->value[0] === "'") {
                         $property_key = $property_branch->value;
+                    } else {
+                        throw new TypeParseTreeException(
+                            ':: in array key is only allowed for ::class',
```

Comment:
that message should be updated now that we support multiple cases. Should we list what is not yet supported instead?

Reply:
No, we still do not support '::' in shape _keys_, even though we do support it now in shape _value_ types (see https://github.com/vimeo/psalm/issues/10669#issuecomment-1933287962), so the message is still correct.

Reply:
Also, see these passing tests below:
https://github.com/vimeo/psalm/pull/10678/files#diff-5e43e612fcbdb0af385d5f84675e4f7d04415e1c607c8d81873fb4914f26309dR364-R373